File: chrome/test/data/webui/chromeos/edu_coexistence/edu_coexistence_browsertest.js
// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/** @fileoverview Runs the EDU Coexistence flow tests. */

GEN_INCLUDE(['//chrome/test/data/webui/chromeos/polymer_browser_test_base.js']);

GEN('#include "content/public/test/browser_test.h"');
GEN('#include "ash/constants/ash_features.h"');

const EduCoexistenceTest = class extends PolymerTest {};

const tests = [
  ['App', 'edu_coexistence_app_test.js'],
  ['Controller', 'edu_coexistence_controller_test.js'],
  ['Ui', 'edu_coexistence_ui_test.js'],
];

tests.forEach(test => registerTest(...test));

/*
 * Add a `caseName` to a specific test to disable it i.e. 'DISABLED_All'
 * @param {string} testName
 * @param {string} module
 * @param {string} caseName
 */
function registerTest(testName, module, caseName) {
  const className = `EduCoexistence${testName}`;
  this[className] = class extends EduCoexistenceTest {
    /** @override */
    get browsePreload() {
      return `chrome://chrome-signin/test_loader.html` +
          `?module=chromeos/edu_coexistence/${module}`;
    }
  };
  TEST_F(className, caseName || 'All', () => mocha.run());
}

File: chrome/test/data/webui/chromeos/edu_coexistence/edu_coexistence_browsertest.cc
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "ash/constants/ash_switches.h"
#include "base/command_line.h"
#include "base/strings/strcat.h"
#include "base/test/scoped_command_line.h"
#include "base/test/scoped_feature_list.h"
#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "chromeos/ash/components/standalone_browser/standalone_browser_features.h"
#include "content/public/test/browser_test.h"

class EduCoexistenceMochaTest : public WebUIMochaBrowserTest {
 protected:
  EduCoexistenceMochaTest() {
    set_test_loader_host(chrome::kChromeUIChromeSigninHost);
  }

  void RunEduCoexistenceTest(const std::string& test_path) {
    RunTest(base::StrCat({"chromeos/edu_coexistence/", test_path}),
            "mocha.run()");
  }
};

IN_PROC_BROWSER_TEST_F(EduCoexistenceMochaTest, App) {
  RunEduCoexistenceTest("edu_coexistence_app_test.js");
}

IN_PROC_BROWSER_TEST_F(EduCoexistenceMochaTest, Controller) {
  RunEduCoexistenceTest("edu_coexistence_controller_test.js");
}

IN_PROC_BROWSER_TEST_F(EduCoexistenceMochaTest, Ui) {
  RunEduCoexistenceTest("edu_coexistence_ui_test.js");
}
